datacube/index/postgis/_products.py

```diff
@@ -385,3 +385,11 @@ def spatial_extent(self, product: str | Product, crs: CRS = CRS("EPSG:4326")) ->
         ids = [ds.id for ds in self._index.datasets.search(product=product.name)]
         with self._db_connection() as connection:
             return connection.spatial_extent(ids, crs)
+
+    def most_recent_change(self, product: str | Product) -> datetime.datetime:
```

Comment:
Seems like this new method should be in abstract.py?

Would require implementations in null and memory, but both should be fairly trivial.